### PR TITLE
Brazilian Portuguese Translation

### PR DESCRIPTION
**Pull request by fermcg**
Brazilian Portuguese Translation v1.0.

### CODE REVIEW

**Comment by fermcg** [comment]
Brazilian Portuguese Translation v1.2
Text improved to enhance understanding
Many terms normalized
Some errors fixed

**Comment by brunobord** [comment]
A few suggestions:

Since this is the first published translation of TBH in Portuguese, maybe you can tag it as "1.0" version. When we'll merge it into master, it'll be *published*. Further fixes may be marked as 1.1 or 1.0.1 if needed.

A few things are missing if we follow the [CONTRIBUTING](https://github.com/brunobord/the-black-hack/blob/master/.github/CONTRIBUTING.md#when-you-add-a-translation) document. That'd be great if you could have these ones checked:

- [ ] You **must** add your contribution to the LICENSE file, in the last section.
- [ ] You *may* add a line in the changelog describing your contribution.
- [ ] Have a look at the section [Specific styles](https://github.com/brunobord/the-black-hack/blob/master/README.md#specific-styles). If you don't know how to handle it, leave it away, I will work this out.

Eventually, when you're completely sure that you're ready to merge, you may want to rebase and squash all these commits into one. This step is not mandatory, it's just to have a cleaner history, with less commits and make it more readable. If you can't or don't want to, don't worry, your PR will still be accepted

**Comment by fermcg** [comment]
Hi,

LICENSE file -> ok
changelog -> ok
converting saves ->  I don't know if it's right, but I added things to
base.html and test-slugify.py and it should work. I can't make html here (I
don't have 'make' in my home computer), but usually when I submit a pull
request there's something there on git hub that does it.

About version 1.0, you're talking about numbering the version in my fork
with this number right? Like folowing theses? https://help.github.com/
articles/creating-releases/ (I'm a little rusty in git, since I've used it
for some personal projects some time ago and in my day job people STILL use
CVS).

When I click the release pane, it shows your project's releases (which is
currently in 2.2.0). I've never did fork & sync so I'm a little lost here.

I've just made a pull request with things as they are. I guess only this
release part is still pending.

File: toolbox/tests/test_slugify.py [hunk 1]
```diff
@@ -17,9 +17,12 @@ def test_slugify_castellano():
...
 
+def test_slugify_portuguese():
+    assert slugify("O que é isso?", '-') == 'o-que-e-isso'
 
 def test_slugify_converting_saves():
     assert slugify('Converting Saves', '-') == 'converting-saves'
     assert slugify('Conversion des jets de sauvegarde', '-') == 'conversion-des-jets-de-sauvegarde'  # noqa
     assert slugify('Convirtiendo tiradas de salvación', '-') == 'convirtiendo-tiradas-de-salvacion'  # noqa
     assert slugify('セーブの変換', '-') == 'sebunobian-huan'
+    assert slugify('Convertendo jogadas de proteção', '-') == 'convertendo-jogadas-de-protecao' # noqa
```
Comment (brunobord) [inline toolbox/tests/test_slugify.py]: Excellent!

File: templates/base.html [hunk 2]
```diff
@@ -67,7 +67,11 @@
     #sebunobian-huan + p + table + table,
 /* italian */
     #convertire-i-tiri-salvezza + p + table,
-    #convertire-i-tiri-salvezza + p + table + table
+    #convertire-i-tiri-salvezza + p + table + table,
+/* portuguese */
+    #convertendo-jogadas-de-protecao + p + table,
+    #convertendo-jogadas-de-protecao + p + table + table
```
Comment (brunobord) [inline templates/base.html]: looking good. I'll double-check this when I'll try to build the HTML pages and browse the result.

**Comment by brunobord** [comment]
> About version 1.0, you're talking about numbering the version in my fork with this number right?

I'm sorry, I think I confused you. Let me sum this up:

* the **website** version is currently 2.2.0. By the *website* I mean either the original texts, the translations and all the toolkit I have written to build the HTML pages, the CSS stylesheets, etc. Whenever some significant changes are made, I'm upgrading the version for the website, but, for example, the French Translation is not upgraded to 2.2.0, neither the other ones.
* Translators don't have to make a release. I, as a maintainer make a release of the whole thing, and make announcements, etc.
* at the moment, there's no "Portuguese (Brazilian)" release of TBH, since this text is still on this branch, and not (yet) browsable on the SRD website.
* as a consequence, **your translation version** could be 1.0, since it's the first time that it'll be published. So let's say that your 1.0 version *of the Brazilian Portuguese translation* is the latest commit of your fork. You've put a 1.0 version during your process, and then a 1.1. Who will know about this 1.1 version? nobody, because no-one will ever read it on the website, it's "hidden" in your branch.
* if some day somebody fixes your translation, finds a typo, etc, it won't be the 1.0 no more and we may upgrade it. So somebody browsing the website will know that there has been some fixes and improvements **on your text file** since the last time she has read it. This "translation version number" will just reflect that **this specific text** has been updated and inform the reader about this.

**Comment by fermcg** [comment]
Yes I guess I've messed things up in the pull request comments. Is there
any way to fix that history?

To make it clear about the version, are we talking about the meta.yaml file?
I've specified 1.2 there because it's based on TBH 1.2 (I've seen the
french version is 1.0).

If it's not the meta, should I use my Readme or write in the text files? (
the-black-hack.md / additional-things.md)

**Comment by brunobord** [comment]
>  Is there any way to fix that history? 
> To make it clear about the version, are we talking about the meta.yaml file?

Simply edit the meta.yaml file and put the right version, that'll be enough.

> To make it clear about the version, are we talking about the meta.yaml file? I've specified 1.2 there because it's based on TBH 1.2 (I've seen the french version is 1.0).

Yes, TBH version used is 1.2, and it's written in your `brazilian-portuguese/the-black-hack.md``, so that's fine. The translation versions and the original text versions don't have to match.

**Comment by fermcg** [comment]
Ok done!

**Comment by brunobord** [comment]
that's cool! thank you for your great work (and your patience with my nutty-nitpicks). Will merge tonight (CEST timezone)

*Obrigado por tudo*

**Comment by fermcg** [comment]
thank you too for having this so well organized! the nutty-nitpicks are
part of the process! Looking forward for the merge!

*Que le Hack se propager à travers le monde*

**Comment by fermcg** [comment]
\o/

**Pull request merged**